src/js/webSerial.js

```diff
@@ -142,8 +142,9 @@ class WebSerial extends EventTarget {
             this.bytesSent = 0;
             this.failed = 0;
             this.openRequested = false;
-
-            this.port.addEventListener("disconnect", this.handleDisconnect.bind(this));
+            if(this.port.addEventListener){
+                this.port.addEventListener("disconnect", this.handleDisconnect.bind(this));
+            }
```

Comment:
Is this a bug? Why make this conditional as this removes disconnect event listener if true.

Reply:
in the polyfill, the SerialPort class doesn't implement EventTarget, and the disconnent event will only be emitted by the Serial class.
In the configurator code, webSerial listens on both, however, it might not be essential. as the gui code would call disconnect in the disconnect event callback from the Serial class.

Reply:
without the test, there will be an exception thrown here.